File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.py
class Solution(object):
    def maxProfit(self, prices):
        n=len(prices)
        ans=0
        min_price=prices[0]

        for i in range(1,n):
            current_profit=prices[i]-min_price
            ans=max(current_profit,ans)
            min_price=min(min_price,prices[i])

        return ans 
        

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int mincost=prices[0];
        int profit=0;
        for(int i=1;i<prices.length;i++){
            int cost=prices[i]-mincost;
            profit=Math.max(cost,profit);
            mincost=Math.min(prices[i],mincost);
        }
        return profit;
        
    }
}